js/reference.js

```diff
@@ -2127,25 +2151,103 @@
                     if (i != 0 && new Reference(module.parse(baseUri + currentPath + (i != 0 ? ';' : '') + filter), self.table.schema.catalog).contextualize.compact._getReadPath().value.length > module.URL_PATH_LENGTH_LIMIT) {
-                        referencePaths.push(currentPath);
+                        // any more filters will go over the url length limit so save the current path and count
+                        // then clear both to start creating a new path
+                        referencePathObjs.push({
+                            path: baseUri + currentPath,
+                            count: currentCount
+                        });
+
                         currentPath = compactPath;
+                        currentCount = 0;
                     } else if (i != 0) {
                         // prepend the conjunction operator when it isn't the first filter to create and we aren't dealing with a url length limit
                         filter = ";" + filter;
                     }
 
+                    // append the filter either on the previous path after adding ";", or on the new path started from compactPath
                     currentPath += filter;
+                    currentCount++;
                 }
-                referencePaths.push(currentPath);
-
-                referencePaths.forEach(function (path) {
-                    var reference = new Reference(module.parse(baseUri + path), self.table.schema.catalog);
-                    references.push(reference);
+                // After last iteration of loop, push the current path
+                referencePathObjs.push({
+                    path: baseUri + currentPath,
+                    count: currentCount
                 });
 
-                return references;
+                var k = 0;
+                var successResponses = [],
+                    deleteErrors = [];
+
+                recursiveDelete(referencePathObjs[k], self.table.schema.catalog);
+
+                return defer.promise;
             } catch (e) {
-                throw module.responseToError(e, self);
+                return module._q.reject(module.responseToError(e, self));
+            }
+
+            function recursiveDelete(pathObj, catalogObj) {
+                var delFlag = module._operationsFlag.DELETE;
+
+                // TODO: expand logging information
+                var headers = {};
+                headers[module.contextHeaderName] = {"action": "delete"};
+                var config = {
+                    headers: headers
+                };
+
+                // NOTE: - if we can't trust the url, make a get to loc.ermrestCompactUri to verify the data set is the same (iterate over keys)
+                //       - a further verification is to take etag of read and use as the if-match header as part of delete to make sure the fetched set is the same as the delete set
+                module._http.delete(pathObj.path, config).then(function (res) {
```

Comment:
You should use the http object that is under `_server` in the Reference (take a look at Reference.delete or any other places that we're calling http module).

This will ensure the retry logic and all the other http module logic are used.